fastestimator/dataset/interleave_dataset.py

```diff
@@ -0,0 +1,271 @@
+        """Split this dataset into multiple smaller datasets.
+
+        This function enables several types of splitting:
+        1. Splitting by fractions.
+            ```python
+            ds = fe.dataset.FEDataset(...)  # len(ds) == 1000
+            ds2 = ds.split(0.1)  # len(ds) == 900, len(ds2) == 100
+            ds3, ds4 = ds.split(0.1, 0.2)  # len(ds) == 630, len(ds3) == 90, len(ds4) == 180
+            ```
+        2. Splitting by counts.
+            ```python
+            ds = fe.dataset.FEDataset(...)  # len(ds) == 1000
+            ds2 = ds.split(100)  # len(ds) == 900, len(ds2) == 100
+            ds3, ds4 = ds.split(90, 180)  # len(ds) == 630, len(ds3) == 90, len(ds4) == 180
+            ```
+        3. Splitting by indices.
+            ```python
+            ds = fe.dataset.FEDataset(...)  # len(ds) == 1000
+            ds2 = ds.split([87,2,3,100,121,158])  # len(ds) == 994, len(ds2) == 6
+            ds3 = ds.split(range(100))  # len(ds) == 894, len(ds3) == 100
+            ```
+
+        Args:
+            *fractions: Floating point values will be interpreted as percentages, integers as an absolute number of
+                datapoints, and an iterable of integers as the exact indices of the data that should be removed in order
+                to create the new dataset.
+            seed: The random seed to use when splitting the dataset. Useful if you want consistent splits across
+                multiple experiments. This isn't necessary if you are splitting by data index.
+            stratify: A class key within the dataset with which to stratify the split (to approximately maintain class
+                balance ratios before and after a split). Incompatible with data index splitting.
+
+        Returns:
+            One or more new datasets which are created by removing elements from the current dataset. The number of
+            datasets returned will be equal to the number of `fractions` provided. If only a single value is provided
+            then the return will be a single dataset rather than a list of datasets.
+
+        Raises:
+            NotImplementedError: If the user created this dataset using one or more non-FEDataset inputs.
+        """
+        if not self.all_fe_datasets:
+            raise NotImplementedError(
+                "InterleaveDataset.split() is not supported when InterleaveDataset contains non-FEDataset objects")
+        # Only pass the stratify argument to the dataset(s) which have the appropriate key
+        new_datasets = [
+            to_list(ds.split(*fractions, seed=seed, stratify=stratify if stratify in ds[0] else None))
+            for ds in self.datasets
+        ]
+        num_splits = len(new_datasets[0])
+        new_datasets = [[ds[i] for ds in new_datasets] for i in range(num_splits)]
+        results = [InterleaveDataset(ds, pattern=self.pattern) for ds in new_datasets]
+        if seed is not None:
+            [ds.fe_reset_ds(seed=seed) for ds in results]
```

Comment:
I think that codacy is flagging this because it's constructing a list of Nones, which is an  unnecessary allocation compared to writing this as a classic for loop

Reply:
alright now it's a classic for loop